File: app/src/main/java/adapters/WebsiteHolder.java
package adapters;

import android.content.Intent;
import android.net.Uri;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.storage.FirebaseStorage;
import com.google.firebase.storage.StorageReference;
import com.sqube.tipshub.R;

import de.hdodenhof.circleimageview.CircleImageView;
import models.ProfileShort;
import models.Website;

public class WebsiteHolder extends RecyclerView.ViewHolder {
    private ImageView imgIcon;
    private TextView txtName;

    WebsiteHolder(@NonNull View itemView) {
        super(itemView);
        imgIcon = itemView.findViewById(R.id.imgIcon);
        txtName = itemView.findViewById(R.id.txtName);
    }

    public void setDisplay(Website website){
        txtName.setText(website.getName());
        StorageReference storageReference = FirebaseStorage.getInstance().getReferenceFromUrl(website.getIcon());
        storageReference.getDownloadUrl().addOnSuccessListener(uri -> {
            Log.i("PeopleAdapter", "onComplete: "+ uri.toString());
            try {
                Glide.with(imgIcon.getContext()).load(uri.toString()).fitCenter().into(imgIcon);
            }
            catch (Exception e){
                Log.w("{PeopleAdapter", "imgIcon GlideApp: " + e.getMessage());
            }
        });
        itemView.setOnClickListener(view -> {
            itemView.getContext().startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(
                    website.getLink())));
        });
    }
}

File: app/src/main/java/adapters/WebsiteHolder.kt
package adapters

import android.net.Uri
import android.util.Log
import androidx.browser.customtabs.CustomTabsIntent
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.google.firebase.storage.FirebaseStorage
import com.sqube.tipshub.R
import com.sqube.tipshub.databinding.ItemSportSiteBinding
import com.sqube.tipshub.models.Website

class WebsiteHolder internal constructor(private val binding: ItemSportSiteBinding) : RecyclerView.ViewHolder(binding.root) {
    private val customTab = CustomTabsIntent.Builder()
            .setToolbarColor(binding.root.context.resources.getColor(R.color.colorPrimary))
            .build()
    fun setDisplay(website: Website) {
        with(binding){
            txtName.text = website.name
            root.setOnClickListener { customTab.launchUrl(root.context, Uri.parse(website.link)) }
            val storageReference = FirebaseStorage.getInstance().getReferenceFromUrl(website.icon)
            storageReference.downloadUrl.addOnSuccessListener { uri: Uri ->
                Log.i("PeopleAdapter", "onComplete: $uri")
                try {
                    Glide.with(imgIcon.context).load(uri.toString()).fitCenter().into(imgIcon)
                } catch (e: Exception) {
                    Log.w("{PeopleAdapter", "imgIcon GlideApp: " + e.message)
                }
            }
        }
    }

}